agent/lib/kontena/workers/service_pod_manager.rb

```diff
@@ -44,10 +44,7 @@ def populate_workers_from_master
       exclusive {
         request = rpc_client.request("/node_service_pods/list", [node.id])
         response = request.value
-        unless response['service_pods'].is_a?(Array)
-          warn "failed to get list of service pods from master: #{response['error']}"
-          return
-        end
+        raise "Invalid response" unless response['service_pods'].is_a?(Array)
```

Comment:
It lets the manager actor crash and restart. Are there expected to be any RPC errors that we can recover from in any more useful way?